RickLeinecker/Fortuna
# No more API calls when switching listings

**Description**
In Marketplace, we were making an API call and filtering all of the sales every time a user clicked a new listings view type. Now, it will get all of the market sales and split them up on the front end. Then, it will swap them out when a new listings view type is clicked. This PR should drastically speed up marketplace, as well as prevent crashes from swapping between types.

The Casus editor CSS change does not alter anything. The master listings header has been updated as well. Also, I fixed the crash on selling last component.

**Resolves These Issues**
Resolves #474 
Resolves #507 

**Type of change**
Check options that are relevant.


**Steps to Verify Functionality**
1. Make sure marketplace has tank sales and component sales.
2. Rapidly switch between them to make sure no crashes occur.
3. Purchase items to ensure that purchasing is still functioning as intended.

**Final Checklist:**
Go down the list and check them off.
